TodoListManagerTest/src/test/java/PerformanceTest/ProjectsPerformanceTest.java

```diff
@@ -0,0 +1,203 @@
+            long start = System.nanoTime();
+            Response response = RestAssured.given().contentType(ContentType.JSON).body(body.toString()).put(pUrl);
+            long end = System.nanoTime();
+            long time = end - start;
+            double time_in_second = (double) time / 1_000_000_000;
+            double cpuUsage = osBean.getCpuLoad() * 100;
+            long memory = osBean.getFreeMemorySize()/ (1024L * 1024L);
+            if(i == target[index]-1){
+                timeStore[index] = time_in_second;
+                cpuUsageStore[index] = cpuUsage;
+                freeMemoryStore[index] = memory;
+                index++;
+            }
+        }
+        System.out.println("---------Add Project Statistics---------");
+        System.out.printf("%-10s %-20s %-20s %-20s%n", "SIZE", "TIME (s)", "CPU USAGE (%)", "MEMORY (MB)");
+        for (int i = 0; i < target.length; i++) {
+            System.out.printf("%-10d %-20.8f %-20.4f %-20d%n",
+                    target[i],
+                    timeStore[i],
+                    cpuUsageStore[i],
+                    freeMemoryStore[i]);
+        }
+        System.out.println("----------------------------------------");
+    }
+
+    @Test
+    public void amendProjectsJsonPost(){
+        double[] timeStore = new double[target.length];
+        double[] cpuUsageStore = new double[target.length];
+        long[] freeMemoryStore = new long[target.length];
+
+        for(int i=0;i<tot;i++){
+            JSONObject body = new JSONObject();
+            String newTitle = "New project " + i;
+            Boolean newCompleted = false;
+            Boolean newActive = false;
+            String newDescription = "New Description";
+            body.put("title",newTitle);
+            body.put("completed",newCompleted);
+            body.put("active",newActive);
+            body.put("description",newDescription);
+            Response response = RestAssured.given().contentType(ContentType.JSON).body(body.toString()).post(pUrl);
+        }
+
+        int index = 0;
+
+        for(int i=0;i<tot;i++){
+            JSONObject body = new JSONObject();
+            String newTitle = "modified project ";
+            Boolean newCompleted = true;
+            Boolean newActive = true;
+            String newDescription = "Modified Description";
+            body.put("title",newTitle);
+            body.put("completed",newCompleted);
+            body.put("active",newActive);
+            body.put("description",newDescription);
+            long start = System.nanoTime();
+            Response response = RestAssured.given().contentType(ContentType.JSON).body(body.toString()).post(pUrl+"/"+i);
+            long end = System.nanoTime();
+            long time = end-start;
+            double time_in_second = (double) time / 1_000_000_000;
+            double cpuUsage = osBean.getCpuLoad() * 100;
+            long memory = osBean.getFreeMemorySize()/ (1024L * 1024L);
```

Comment:
logic is wrong here, you are populating everthing and then you modify, but the correct step should be amending when populating.

Reply:
Let me have a look.